ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh

```diff
@@ -159,19 +157,65 @@ if [[ ! -z "$OLD_CLUSTER" ]]; then
 fi
 
 # Get the openshift version
-version_cmd="rosa list versions --channel-group ${CHANNEL_GROUP} -o json"
-if [[ "$HOSTED_CP" == "true" ]]; then
-  version_cmd="$version_cmd --hosted-cp"
+if [[ -z "${CHANNEL_GROUPS}" ]]; then
+  log "Error: CHANNEL_GROUPS variable is empty"
+  exit 1
 fi
-if [[ ${AVAILABLE_UPGRADE} == "yes" ]] ; then
-  # shellcheck disable=SC2089
-  version_cmd="$version_cmd | jq -r '.[] | select(.available_upgrades!=null) .raw_id'"
-else
-  version_cmd="$version_cmd | jq -r '.[].raw_id'"
+IFS=',' read -r -a CHANNEL_GROUPS_ARRAY <<< "${CHANNEL_GROUPS// /}"
+
+CHANNEL_GROUP=""
+
+for GROUP in "${CHANNEL_GROUPS_ARRAY[@]}"; do
+  log "Checking ${GROUP} channel..."
+
+  version_cmd="rosa list versions --channel-group ${GROUP} -o json"
+  if [[ "${HOSTED_CP}" == "true" ]]; then
+    version_cmd="${version_cmd} --hosted-cp"
+  fi
+
+  if [[ "${AVAILABLE_UPGRADE}" == "yes" ]]; then
+    versionList=$(eval "${version_cmd}" | jq -r '.[] | select(.available_upgrades!=null) .raw_id')
```

Comment:
All the `versionList` vars should match the same format as the others, i.e., `version_list`.